binding_generator.py

```diff
@@ -2719,6 +2719,8 @@ def correct_type(type_name, meta=None, use_alias=True):
     if meta is not None:
         if "int" in meta:
             return f"{meta}_t"
+        elif "char" in meta:
+            return f"{meta}_t"
         elif meta in type_conversion:
             return type_conversion[type_name]
```

Comment:
Looking at the `type_conversion` above and
```
	static const char *argmeta[13] = { "none", "int8", "int16", "int32", "int64", "uint8", "uint16", "uint32", "uint64", "float", "double", "char16", "char32" };
```
does this mean that `float` meta in the above array also gets converted to `double`?

Reply:
Yes, that's unfortunate. I think this `elif` can be removed since you are never going to get `int` or `Nil` as metadata, and if you get `float` then it means `float` (not `double`).